python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,330 @@
+
+
+    async def wait_for_chiller_temp_within_tolerance(self):
+        chiller_temp_within_tolerance = False
+        while time.time() - start_time_chill_time < self.timeout_chiller_cool_down:
+            chiller_temps = await white_light_source.tel_chillerTemperatures.aget()
+            tel_chiller_temp = chiller_temps.supplyTemperature
+            if (
+                chiller_temps.setTemperature
+                * (1.0 - self.chiller_temp_tolerance_relative)
+                < tel_chiller_temp
+                < chiller_temps.setTemperature
+                * (1.0 + self.chiller_temp_tolerance_relative)
+            ):
+                chiller_temp_within_tolerance = True
+                chill_time = time.time() - start_time_chill_time
+                self.log.info(
+                    f"Chiller reached target temperature {tel_chiller_temp} within tolerance in {chill_time} s"
+                )
+                break
+            else:
+                continue
+        else:
+            self.log.info(
+                f"Gave up waiting after {self.chiller_temp_time} for the chiller to chill to {self.chiller_temperature}. Stayed at {tel_chiller_temp}") 
+
+
+    # async def open_white_light_shutter(self):
+    #     await self.white_light_source.cmd_openShutter.set_start()
+    #     ## Should we include a check that the shutter is open or not necessary?
+
+    async def switch_lamp_on(self):
+        await self.white_light_source.cmd_turnLampOn.set_start(
+            power=self.whitelight_power, timeout = self.timeout_lamp_warm_up
+        )
+
+        # # Confirm lamp turns on (It will go into a warm up period before it will turn on)
+        # try:
+        #     await asyncio.wait_for(
+        #         self.has_lamp_finish_warm_up(), self.lamp_warm_up_time
+        #     )
+        # except asyncio.TimeoutError:
+        #     self.log.info(f"Lamp didn't turn on after {self.lamp_warm_up_time} s")
+
+    async def wait_for_lamp_to_warm_up(self):
+        await self.white_light_source.evt_lampState.flush()
+        lamp_state = await self.white_light_source.evt_lampState.aget(
+            timeout=self.cmd_timeout)
+        
+        # lamp_done_warm_up = False
+        start_time_lamp_warm_up = time.time()
+        while lamp_state.basicState != ATWhiteLight.LampBasicState.ON:
+            lamp_state = await self.white_light_source.evt_lampState.next(
+                flush=False,
+                timeout=self.timeout_lamp_warm_up
+            )
+            if lamp_state.basicState == ATWhiteLight.LampBasicState.WARMUP:
+                warmup_time_left = (
+                    lamp_state.warmupEndTime - lamp_state.private_rcvStamp
+                )
+                self.log.info(
+                    "Time Left for lamp warmup: {} min.".format(warmup_time_left / 60.0)
+                )
+                await asyncio.sleep(self.track_lamp_warmup)
+            elif lamp_state.basicState == ATWhiteLight.LampBasicState.ON:
```

Comment:
I think you could drop this `elif` and just do this after the `while` loop has ended (it saves you from doing a comparison).

Reply:
I have changed the code a little more.

Reply:
Sorry, not sure I understand the logic here. Do you need to break in this condition? it seems like it is going to continue in the loop until is time out. 

However, see my suggestion for simplifying this loop.